core/src/main/java/org/springframework/security/authorization/method/AuthorizationManagerBeforeMethodInterceptor.java

```diff
@@ -245,8 +246,8 @@ public void setSecurityContextHolderStrategy(SecurityContextHolderStrategy secur
 
 	private Object attemptAuthorization(MethodInvocation mi) throws Throwable {
 		this.logger.debug(LogMessage.of(() -> "Authorizing method invocation " + mi));
-		AuthorizationDecision decision = this.authorizationManager.check(this::getAuthentication, mi);
-		this.eventPublisher.publishAuthorizationEvent(this::getAuthentication, mi, decision);
+		AuthorizationResult decision = this.authorizationManager.authorize(this::getAuthentication, mi);
+		this.eventPublisher.publishAuthorizationEvent(this::getAuthentication, mi, (AuthorizationDecision) decision);
```

Comment:
Let's consider also adding a `publishAuthorizationEvent` implementation that uses `AuthorizeResult`.

Reply:
I can add a new method:
```
	default <T> void publishAuthorizationEvent(Supplier<Authentication> authentication, T object,
			AuthorizationResult decision) {
		publishAuthorizationEvent(authentication, object,(AuthorizationDecision) decision);
	}
```
Method with `AuthorizationDecision` semantics can be declared as deprecated. This will not break backward compatibility with existing `AuthorizationEventPublisher` implementations

Reply:
@jzheaux I suggest adding an `AuthorizationEventPublisher` implementation that will not publish events, like `NoopAuthorizationEventPublisher`:
```
public final class NoopAuthorizationEventPublisher implements AuthorizationEventPublisher {

	@Override
	public <T> void publishAuthorizationEvent(Supplier<Authentication> authentication, T object, AuthorizationDecision decision) {

	}

	@Override
	public <T> void publishAuthorizationEvent(Supplier<Authentication> authentication, T object, AuthorizationResult decision) {
	}
}
```
`AuthorizationEventPublisher` is not marked as a functional interface, but in fact it is one. `AuthorizationManagerAfterMethodInterceptor` and `AuthorizationFilter` use `noPublish` method to implement the publisher. This means that calling `this.eventPublisher.publishAuthorizationEvent(this::getAuthentication, request, decision);` will require the `AuthorizationDecision` type explicitly. At the same time, we cannot remove the `publishAuthorizationEvent` method with `AuthorizationDecision` semantic. Therefore, it makes sense to replace `eventPublisher` in these components with `NoopAuthorizationEventPublisher`.